R/tags.R

```diff
@@ -687,18 +733,40 @@ findDependencies <- function(tags, tagify = TRUE) {
   if (isTRUE(tagify)) {
     tags <- tagify(tags)
   }
-  dep <- htmlDependencies(tags)
-  if (!is.null(dep) && inherits(dep, "html_dependency"))
-    dep <- list(dep)
+  deps <- resolveFunctionalDependencies(htmlDependencies(tags))
   children <- if (is.list(tags)) {
     if (isTag(tags)) {
       tags$children
     } else {
       tags
     }
   }
-  childDeps <- unlist(lapply(children, findDependencies, tagify = FALSE), recursive = FALSE)
-  c(childDeps, if (!is.null(dep)) dep)
+  childDeps <- unlist(lapply(children, findDependencies, tagify = FALSE), recursive = FALSE, use.names = FALSE)
+  c(childDeps, deps)
+}
+
+#' Resolve tag functions
+#'
+#' Resolves any [tagFunction()]s inside a list of [htmlDependencies()]. To
+#' resolve [tagFunction()]s _and then_ remove redundant dependencies all at once,
+#' use [resolveDependencies()] (which calls this function internally).
+#'
+#' @inheritParams resolveDependencies
+#' @export
+#' @return a list of HTML dependencies.
+#' @seealso [tagFunction()], [resolveDependencies()]
+resolveFunctionalDependencies <- function(dependencies) {
```

Comment:
`dependencies` will be `NULL` in the vast majority of cases, so it makes sense to have a check for that (an early return) at the top of this function.